### DISCUSSION TITLE
Removing signatures from a signed PDF.

### DISCUSSION THREAD

**Discussion by arsultani** [pymupdf/PyMuPDF#1493] [Looking for help] [answered]
Hello,

I absolutely love all the functionalities that Fitz provides! It covers everything I need except one little thing that I have been having trouble with. I use Fitz to annotate a signed PDF and, as expected, the signatures become invalid the moment you modify the PDF. I was wondering if there was a way for me to REMOVE the signatures from the PDF. When a document is signed, whether it's a valid signature or not, the end user cannot manually add pictures to the PDF. They also see the invalid signature warning upon opening the file.

The only way I could do this was to go from PDF to Pixmap to PDF. That way the signatures are lost, but unfortunately so is the text layer. Is there a clean and easy way to do this? I can't seem to find any other way looking through the docs.

Thanks!

**Comment by JorjMcKie** [comment, marked as answer]
Hm, PyMuPDF currently does not support adding signature widgets or signing the PDF.
But otherwise handling other widgets / fields is support. As well as removing fields or resetting their content.

So you should be able to iterate over the pages, look for signed signatures and reset them / or delete the signature field altogether.

If you only want to add new content that should be possible too, as long as you **_save incrementally_** afterwards - which does not invalidate the signature.

**Reply by arsultani** [reply]
Thank you! 

I had missed the `widget` section of the docs. I was wrongly focused on believing that signatures were `annots`. Your comment guided me to the right section. Now I iterated over each signature `widget` in the page and I can `reset()` them to get what I want. I will give the incremental save a try as well.

Happy new year and thank you so much for your work! :)

**Reply by JorjMcKie** [reply]
Wish yo the best as well. Stay healthy!
